python/cudf/cudf/tests/test_pandas_comparison.py

```diff
@@ -0,0 +1,366 @@
+                    f"{{' not' if pandas_modified else ''}}."
+                )
+            """
+        )
+        globals_ = {
+            "test": test,
+            "remove_args": (f"{_LIB_PARAM_NAME}",)
+            if _LIB_PARAM_NAME in parameters
+            else None,
+            "cudf_object": cudf_object,
+            "modify": modify,
+        }
+        exec(src, globals_)
+        wrapped_test = globals_["wrapped_test"]
+        # In case marks were applied to the original benchmark, copy them over.
+        if marks := getattr(test, "pytestmark", None):
+            wrapped_test.pytestmark = marks
+        return wrapped_test
+
+    return deco
+
+
+@pandas_comparison_test
+def test_init(lib):
+    data = [
+        (5, "cats", "jump", np.nan),
+        (2, "dogs", "dig", 7.5),
+        (3, "cows", "moo", -2.1, "occasionally"),
+    ]
+    return lib.DataFrame(data)
+
+
+@pandas_comparison_test()
+def test_init2(lib):
+    data = [
+        (5, "cats", "jump", np.nan),
+        (2, "dogs", "dig", 7.5),
+        (3, "cows", "moo", -2.1, "occasionally"),
+    ]
+    return lib.DataFrame(data)
+
+
+@pytest.fixture
+def dt():
+    return [
+        (5, "cats", "jump", np.nan),
+        (2, "dogs", "dig", 7.5),
+        (3, "cows", "moo", -2.1, "occasionally"),
+    ]
+
+
+@pandas_comparison_test
+def test_init4(lib, dt):
+    return lib.DataFrame(dt)
+
+
+@pandas_comparison_test
+def test_init7(dt, lib):
+    return lib.DataFrame(dt)
+
+
+@pandas_comparison_test
+@pytest.mark.parametrize(
+    "data",
+    [
+        [
+            (5, "cats", "jump", np.nan),
+            (2, "dogs", "dig", 7.5),
+            (3, "cows", "moo", -2.1, "occasionally"),
+        ]
+    ],
+)
+def test_init5(lib, data):
+    return lib.DataFrame(data)
+
+
+@pytest.mark.parametrize(
+    "data",
+    [
+        [
+            (5, "cats", "jump", np.nan),
+            (2, "dogs", "dig", 7.5),
+            (3, "cows", "moo", -2.1, "occasionally"),
+        ]
+    ],
+)
+@pandas_comparison_test
+def test_init6(lib, data):
+    return lib.DataFrame(data)
+
+
+@pytest.mark.parametrize(
+    "df",
+    [
+        pd.DataFrame(
+            [
+                (5, "cats", "jump", np.nan),
+                (2, "dogs", "dig", 7.5),
+                (3, "cows", "moo", -2.1, "occasionally"),
+            ]
+        )
+    ],
+)
+@pandas_comparison_test
+def test_from_pandas(lib, df):
+    return lib.from_pandas(df) if _is_cudf(lib) else df
+
+
+def custom_assert(expected, got):
+    cudf.testing.assert_frame_equal(cudf.from_pandas(expected), got)
+
+
+@pytest.mark.parametrize(
+    "df",
+    [
+        pd.DataFrame(
+            [
+                (5, "cats", "jump", np.nan),
+                (2, "dogs", "dig", 7.5),
+                (3, "cows", "moo", -2.1, "occasionally"),
+            ]
+        )
+    ],
+)
+@pandas_comparison_test(assert_func=custom_assert)
+def test_from_pandas_custom_assert(lib, df):
+    return lib.from_pandas(df) if _is_cudf(lib) else df
```

Comment:
Can this just use a constructor?
```suggestion
    return lib.DataFrame(df)
```